### ISSUE TITLE
reactor fails to run when submitting request to NFS

### ISSUE THREAD

**Issue by tchaikov** [scylladb/seastar#1947]
when running a seastar application and it sends io request to files located on a filesystem which does not support `RWF_NOWAIT`, `io_submit()` fails with `ENOTSUPP`:

```
ERROR 2023-11-11 01:16:53,219 [shard 0] seastar - io_submit: Operation not supported `
exception running reactor main loop on shard 0.
Backtrace:
  0x4ff41f8
  0x5006883
  0x50050b8
  0x4fabaf5
  0x4faac68
  0x1148d34
  0x114a840
  0x114746a
  /opt/scylladb/libreloc/libc.so.6+0x2750f
  /opt/scylladb/libreloc/libc.so.6+0x275c8
  0x1145624
2023-11-11 01:16:53,315 INFO exited: scylla (exit status 1; not expected)
```

it turns out quite a few (if not all) networking filesystem do not support nowait aio, so in addition to nfs, this problem also applies to 9p, cephfs, cifs and ocfs2.

we could print a more informative error message instead of 

> `io_submit: Operation not supported`

and a backtrace

**Comment by hiqsociety** [comment]
1. possible to make seastar enable this?
2. what are the networking filesystems currently supported by seastar?

**Comment by tchaikov** [comment]
> 1. possible to make seastar enable this?

as Nadav put in another thread. it might be possible. but this is not our priority at this moment.

> 2. what are the networking filesystems currently supported by seastar?

as the description of this issue put, AFAICT, no known network filesystem or stacked filesystem are supported. at least the ones in linux 6.3 are not supported.

**Comment by hiqsociety** [comment]
@tchaikov allow me to thank you for this reply. 
sorry for the late response, can i please ask for the version of seastar support that works on nfs and with least amount of bug or caveats to look out for?

i'm using scylladb and if i cant use nfs, it's a no-go for my deployment.

can you provide the scylladb version (with least bug) for production use using nfs as storage backend? i understand the settings needing intr options for nfs 4.1 etc. or maybe you can just provide your expert opinion on how to get this done.

basically i'm prepared to use legacy version of scylladb just for the NFS backend and definitely hope to get your input on what to look out for for production use.


p.s. : i helped operate the english telegram group of scylladb and i'm quite sure there are a few people who would like to know about this too.

appreciate this prompt reply. thx!

**Comment by hiqsociety** [comment]
@tchaikov any feedback on this? what legacy version of scylladb will work with nfs for production use?

**Comment by tchaikov** [comment]
> @tchaikov any feedback on this? what legacy version of scylladb will work with nfs for production use?

not that I'm aware of. but the more appropriate venue for scylladb related question is probably https://github.com/scylladb/scylladb/issues/16069 where you explained your interests in running scylladb on NFS . but i think you already got the same answer there.

**Comment by hiqsociety** [comment]
@tchaikov what version of seastar has nfs support then? i will check the cut off point date to get the version of scylladb that works. can you give an idea?

**Comment by tchaikov** [comment]
i don't think it's ever supported. please see the description of this issue for more details.
